fire/SFMainMod.F90

```diff
@@ -954,18 +955,18 @@ subroutine  crown_damage ( currentSite )
                 endif ! evaluate passive crown fire
                 
                 ! Flames lower than bottom of canopy. 
-                ! c%hite is height of cohort  
-                if (currentPatch%SH < (currentCohort%hite- height_cbb)) .and. currentCohort%crown_fire_flg = 0 then 
+                ! cohort%hite is height of cohort  
+                if (currentPatch%SH <= height_cbb .and. currentCohort%crown_fire_flg = 0) then 
                    currentCohort%fraction_crown_burned = 0.0_r8
                 else
                    ! Flames part of way up canopy. 
                    ! Equation 17 in Thonicke et al. 2010. 
                    ! flames over bottom of canopy but not over top.
-                   if ((currentCohort%hite > 0.0_r8).and.(currentPatch%SH >=  &
-                      (currentCohort%hite - height_cbb))) .and. currentCohort%crown_fire_flg = 0 then 
+                   if ((currentCohort%hite > 0.0_r8).and.(currentPatch%SH > height_cbb) &
+                        .and. (currentPatch%SH <= currentCohort%hite)  &
+                        .and. currentCohort%crown_fire_flg = 0 then 
 
-                      currentCohort%fraction_crown_burned =  (currentPatch%SH-currentCohort%hite*(1.0_r8- &
-                             EDPftvarcon_inst%crown(currentCohort%pft)))/(height_cbb) 
+                      currentCohort%fraction_crown_burned =  (currentPatch%SH - height_cbb)/crown_depth 
```

Comment:
@jkshuman I'm realizing that I prefer the following alternative here:
1) remove line 966:
`.and. (currentPatch%SH <= currentCohort%hite)  &`
2) change line 969 to:
`currentCohort%fraction_crown_burned =  max(0.0_r8, min(1.0_r8, (currentPatch%SH - height_cbb)/crown_depth))`
3) remove the `else  ! Flames over top of canopy.` lines 971-973

The advantage of this is that you make lines 971-973 unnecessary and maintain the relevance of
`currentCohort%hite > 0.0_r8`
`currentPatch%SH > height_cbb`
`currentCohort%crown_fire_flg = 0`
that currently appear in the if statement but not in the else.

So leaving the code as is, you could end up with 
`currentCohort%fraction_crown_burned =  1.0_r8`
for `currentCohort%hite = 0.0`  (probably not a big deal)
or for `currentCohort%crown_fire_flg = 1`  (not a problem at the moment, but...)
The latter would become a problem if we decided to allow passive crown fires to give
`currentCohort%fraction_crown_burned < 1.0_r8`
because then the else would overwrite the value with 1.0_r8.

Last thing...
If you do change these lines as I suggest, then you could clean up further by moving the `if` in lines 965-967 to line 961 and make into an `else if`

Let me know if I'm making sense of if I've confused myself and am completely wrong about this.

Reply:
@slevisconsulting I agree with you on these. made some updates.